### PR TITLE
Load sidecars and narratives

### LINKED ISSUE #7: Allow sidecar files to be loaded in auspice.us
Sidecar files (`_tip-frequencies.json`, `_root-sequence.json`) are additional fetches in auspice. For auspice.us we should allow these files to be dragged on as part of the initial load. It may also be wise to allow them to be dragged on after the main dataset is loaded, but this is not essential.

### LINKED ISSUE #20: [minor] HTML error in splash page
React error (in development mode):

```
 Warning: validateDOMNesting(...): <ul> cannot appear as a descendant of <p>.
    in ul (created by SplashContent)
    in p (created by Context.Consumer)
```

### PR DESCRIPTION
**Pull request by jameshadfield**
See commit messages for details. I’ll post a tarball of files used for testing this on slack. 

The measurements sidecar failed to load, which seemed on the surface to be an auspice error (not auspice.us). We can make this a separate issue perhaps?

Closes #7
Closes #20

### CODE REVIEW

**Review by joverlee521 — CHANGES_REQUESTED** [review]
Glad to see this wasn't _too_ big of change! 

I left some minor inline comments and I also have some general comments:

- I noticed that all "valid" files are read into memory even though we only support displaying "one thing". This can lead to a long wait for the files to be read or memory issues if someone drags on a crazy number of files. We might want to delay reading the files to right before the dispatch of the data so we only read the files that will actually be used. 

- A reminder to update the "Currently supported data formats" section on the splash page.  It would be helpful to also include a comment on which dropped files should be expected to be displayed. For example, a narrative will always take precedence over the datasets and if multiple datasets are added then only two will be displayed as two trees side by side.

File: auspice_client_customisation/handleDroppedFiles.js [hunk 1]
```diff
@@ -1,74 +1,155 @@
...
       }));
+    } 
+  }
+
+  /* are there any files we haven't (attempted to) read? */
+  for (const file of files) {
+    if (!filesSeen.has(file.name.toLowerCase())) {
+      logs.push(`Unparsed file: ${file.name}`);
     }
+  }
```
[the remark this answers is not in the document]
Comment (jameshadfield) [reply auspice_client_customisation/handleDroppedFiles.js:143]: Yup - I wanted to first load the main dataset files (i.e. `Dataset()`s) then add in sidecars to those.

File: auspice_client_customisation/handleDroppedFiles.js [hunk 2]
```diff
@@ -109,6 +100,24 @@ async function collectDatasets(dispatch, files) {
...
+  for (const file of files) {
+    const nameLower = file.name.toLowerCase();
+    if (filesSeen.has(file.name) || !nameLower.endsWith("json")) continue;
+    for (const [sidecarSuffix, sidecarPropName] of Object.entries(sidecarMappings)) {
+      if (nameLower.endsWith(`_${sidecarSuffix}.json`)) { // filename looks like a sidecar file?
+        filesSeen.add(nameLower);
+        const mainNameLower = nameLower.replace(`_${sidecarSuffix}.json`, '.json');
+        if (datasets[mainNameLower]) {
+          datasets[mainNameLower][sidecarPropName] = readFile(file);
+          logs.push(`Read ${file.name} as a sidecar file of ${datasets[mainNameLower].name}`);
```
[the remark this answers is not in the document]
Comment (jameshadfield) [reply auspice_client_customisation/handleDroppedFiles.js:117]: Yes, if we were to to change it to `await readFile(file);`. As it stands,`datasets[mainNameLower][sidecarPropName]` is a promise that will resolve/reject. The promises are used in the [loadSidecars function](https://github.com/nextstrain/auspice/blob/master/src/actions/loadData.js#L305-L331), and most of them have a `catch` handler to deal with a rejected promise.

File: auspice_client_customisation/handleDroppedFiles.js [hunk 3]
```diff
@@ -1,74 +1,256 @@
...
+        message: "Failed to parse additional metadata file!",
+        details: "Please drop the metadata file after the tree has loaded."
       }));
+    } 
+  }
+
+  /* loop through files and, if a sidecar, load it into the associated `Dataset` object */
+  for (const file of files) {
+    const nameLower = file.name.toLowerCase();
+    if (filesSeen.has(file.name) || !nameLower.endsWith("json")) continue;
```
[the remark this answers is not in the document]
Comment (jameshadfield) [reply auspice_client_customisation/handleDroppedFiles.js:110]: Sure should be! Fixed

**Comment by jameshadfield** [comment]
Thanks @joverlee521!

> I noticed that all "valid" files are read into memory even though we only support displaying "one thing". This can lead to a long wait for the files to be read or memory issues if someone drags on a crazy number of files. 

This might be possible, but in practice I think it'll be rarely encountered. As you point out, there are tricks we can use if we need to address this situation in the future (unless you've experienced this in testing?). For what it's worth, I think dropping multiple datasets onto auspice.us isn't very useful, except in the case of narratives. 

>  A reminder to update the "Currently supported data formats" section on the splash page. 

Thanks for the reminder! Done in [86a1a34](https://github.com/nextstrain/auspice.us/pull/23/commits/86a1a34049722fe32b7b05ec624cb85c8e4d2e38)

File: auspice_client_customisation/handleDroppedFiles.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (joverlee521) [reply auspice_client_customisation/handleDroppedFiles.js:117]: Ah right, totally missed these get used in the loadSidecars functions. That makes sense!

**Comment by joverlee521** [comment]
> This might be possible, but in practice I think it'll be rarely encountered. As you point out, there are tricks we can use if we need to address this situation in the future (unless you've experienced this in testing?).

Yeah, I don't see people doing this but just wanted to point it out. When I was testing, I dropped all files from the auspice `get-data` command just to see what would happen. It took a little longer to load but nothing crashed.

**Pull request merged**